pkg/manifests/manifests.go

```diff
@@ -2079,6 +2091,75 @@ func (f *Factory) PrometheusAdapterAPIService() (*apiregistrationv1.APIService,
 	return f.NewAPIService(f.assets.MustNewAssetReader(PrometheusAdapterAPIService))
 }
 
+func (f *Factory) MetricsServerServiceAccount() (*v1.ServiceAccount, error) {
+	return f.NewServiceAccount(f.assets.MustNewAssetReader(MetricsServerServiceAccount))
+}
+
+func (f *Factory) MetricsServerClusterRole() (*rbacv1.ClusterRole, error) {
+	return f.NewClusterRole(f.assets.MustNewAssetReader(MetricsServerClusterRole))
+}
+
+func (f *Factory) MetricsServerClusterRoleBinding() (*rbacv1.ClusterRoleBinding, error) {
+	return f.NewClusterRoleBinding(f.assets.MustNewAssetReader(MetricsServerClusterRoleBinding))
+}
+
+func (f *Factory) MetricsServerClusterRoleAggregatedMetricsReader() (*rbacv1.ClusterRole, error) {
+	return f.NewClusterRole(f.assets.MustNewAssetReader(MetricsServerClusterRoleAggregatedMetricsReader))
+}
+
+func (f *Factory) MetricsServerClusterRoleBindingAuthDelegator() (*rbacv1.ClusterRoleBinding, error) {
+	return f.NewClusterRoleBinding(f.assets.MustNewAssetReader(MetricsServerClusterRoleBindingAuthDelegator))
+}
+
+func (f *Factory) MetricsServerRoleBindingAuthReader() (*rbacv1.RoleBinding, error) {
+	return f.NewRoleBinding(f.assets.MustNewAssetReader(MetricsServerRoleBindingAuthReader))
+}
+
+func (f *Factory) MetricsServerDeployment() (*appsv1.Deployment, error) {
```

Comment:
The TLS settings should be injected at runtime based on the API global settings like we do here for prometheus-adapter
https://github.com/openshift/cluster-monitoring-operator/blob/ebaa7e3bc9fd26ed8566a046d29a61d4e9db6c3e/pkg/manifests/manifests.go#L2013-L2014